File: langs/java/problems/text_editor/stub.java
public class Simulation {
    public Simulation() {}

    /**
     * Insert at pos. New length, or {@code invalid_request}.
     */
    public String insert(int pos, String text) {
        return "";
    }

    /**
     * Deleted substring, or {@code invalid_request}.
     */
    public String erase(int pos, int n) {
        return "";
    }

    /**
     * Full buffer.
     */
    public String getText() {
        return "";
    }

    /**
     * Length as a string.
     */
    public String length() {
        return "";
    }

    /**
     * Set cursor. {@code "true"} or {@code invalid_request}.
     */
    public String move(int pos) {
        return "";
    }

    /**
     * Insert at cursor and advance. New length.
     */
    public String typeText(String text) {
        return "";
    }

    /**
     * Current position as a string.
     */
    public String cursor() {
        return "";
    }

    /**
     * Revert last insert/erase/typeText. {@code "true"} or {@code "false"}.
     */
    public String undo() {
        return "";
    }

    /**
     * Reapply an undone op. {@code "true"} or {@code "false"}.
     */
    public String redo() {
        return "";
    }

    /**
     * Mark {@code [start, end)}. {@code "true"} or {@code invalid_request}.
     */
    public String select(int start, int end) {
        return "";
    }

    /**
     * Delete selection onto clipboard, or {@code invalid_request}.
     */
    public String cut() {
        return "";
    }

    /**
     * Copy selection onto clipboard, or {@code invalid_request}.
     */
    public String copySel() {
        return "";
    }

    /**
     * Insert clipboard at cursor. New length, or {@code invalid_request}.
     */
    public String paste() {
        return "";
    }
}

File: langs/python3/problems/text_editor/stub.py
class Simulation:
    def __init__(self):
        pass

    def insert(self, pos, text):
        """Insert at pos. New length, or invalid_request."""
        raise NotImplementedError

    def erase(self, pos, n):
        """Deleted substring, or invalid_request."""
        raise NotImplementedError

    def get_text(self):
        """Full buffer."""
        raise NotImplementedError

    def length(self):
        """Length as a string."""
        raise NotImplementedError

    def move(self, pos):
        """Set cursor. "true" or invalid_request."""
        raise NotImplementedError

    def type_text(self, text):
        """Insert at cursor and advance. New length."""
        raise NotImplementedError

    def cursor(self):
        """Current position as a string."""
        raise NotImplementedError

    def undo(self):
        """Revert last insert/erase/type_text. "true" or "false"."""
        raise NotImplementedError

    def redo(self):
        """Reapply an undone op. "true" or "false"."""
        raise NotImplementedError

    def select(self, start, end):
        """Mark [start, end). "true" or invalid_request."""
        raise NotImplementedError

    def cut(self):
        """Delete selection onto clipboard, or invalid_request."""
        raise NotImplementedError

    def copy_sel(self):
        """Copy selection onto clipboard, or invalid_request."""
        raise NotImplementedError

    def paste(self):
        """Insert clipboard at cursor. New length, or invalid_request."""
        raise NotImplementedError
